### PR TITLE
Add build step for publishing

### PR DESCRIPTION
**Pull request by mrodm**
## Proposed commit message

This PR adds the steps to trigger sign and publish in Buildkite

- Example of build where it was forced to publish a package https://buildkite.com/elastic/integrations/builds/7023

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- ~[ ] I have verified that all data streams collect metrics or logs.~
- ~[ ] I have added an entry to my package's `changelog.yml` file.~
- ~[ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).~

## Author's Checklist

- [x] Remove changes from `.ci/Jenkinsfile`
- [x] Test to publish a package with DRY_RUN=true https://buildkite.com/elastic/integrations/builds/7023
- [x] Force for now to not publish now adding a DRY_RUN variable as it was done in Jenkins step

## Related issues

-

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -29,10 +29,10 @@ steps:
       cpu: "8"
       memory: "4G"
 
-  # TODO: Pending to migrate steps to Buildkite pipelines - still triggering Jenkins pipelines
-  - label: "Publish packages - (triggering Jenkins builds)"
+  # TODO: To be removed Pending to migrate steps to Buildkite pipelines - still triggering Jenkins pipelines
+  - label: ":jenkins: Publish packages - (triggering Jenkins builds)"
```
Comment (mrodm) [inline .buildkite/pipeline.yml:33]: Should we remove in this PR this step that performs the publication using Jenkins pipelines? It has not been used never in Buildkite.

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -41,6 +41,25 @@ steps:
...
+      ARTIFACTS_FOLDER: "artifacts-to-sign"
+      DRY_RUN: "true"  # TODO: to be removed
+    depends_on:
+      - step: "check"
+        allow_failure: false
+    artifact_paths:
+      - artifacts-to-sign/*.zip
+
+  # TODO: Remove force to just have the package and signature files as artifacts when signing pipeline is triggered
+  - wait
```
Comment (mrodm) [inline .buildkite/pipeline.yml:61]: This is something that we can remove maybe.

Would it be interesting to have the publication process run in parallel with the testing ?

Comment (mrodm) [reply .buildkite/pipeline.yml:61]: Ok, I'll remove this `wait` step so the steps related to publication and all the steps for testing all packages are run in parallel.

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -29,17 +29,21 @@ steps:
...
+    key: "build-packages"
+    command: ".buildkite/scripts/build_packages.sh"
     agents:
-      provider: "gcp"
+      image: "${LINUX_AGENT_IMAGE}"
+      cpu: "8"
+      memory: "8G"
+    env:
+      ARTIFACTS_FOLDER: "artifacts-to-sign"
+      DRY_RUN: "true"  # TODO: to be removed
```
Comment (mrodm) [inline .buildkite/pipeline.yml:41]: I'll create another specific PR to enable this step in Buildkite and disable the step in Jenkins

**Review by sharbuz — APPROVED** [review]

**Pull request merged**